File: spring_java/src/main/java/dev/pollito/spring_java/sakila/film/domain/model/Film.java
package dev.pollito.spring_java.sakila.film.domain.model;

import static lombok.AccessLevel.*;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.jspecify.annotations.NonNull;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = PRIVATE)
public class Film {
  Integer id;
  @NonNull String title;
  String description;
  Integer releaseYear;
  FilmRating rating;
  Integer length;
  @NonNull FilmLanguage language;
  FilmLanguage originalLanguage;
  @NonNull Integer rentalDuration;
  @NonNull BigDecimal rentalRate;
  @NonNull BigDecimal replacementCost;
  String specialFeatures;
  OffsetDateTime lastUpdate;
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/film/domain/model/Film.kt
package dev.pollito.spring_kotlin.sakila.film.domain.model

import java.math.BigDecimal
import java.time.OffsetDateTime

data class Film(
    val id: Int?,
    val title: String,
    val description: String?,
    val releaseYear: Int?,
    val rating: FilmRating?,
    val length: Int?,
    val language: FilmLanguage,
    val originalLanguage: FilmLanguage?,
    val rentalDuration: Int,
    val rentalRate: BigDecimal,
    val replacementCost: BigDecimal,
    val specialFeatures: String?,
    val lastUpdate: OffsetDateTime?,
)
